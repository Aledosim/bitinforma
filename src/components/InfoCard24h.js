import React from 'react';

import styles from '../styles/components/InfoCard24h.module.css'

export default function InfoCard24h(){
    return(
        <div id='infocard24h' className={styles.infoCard24hContainer}>
            <header>
            </header>
            <div>
                <p>
                    Maior preço negociado:
                </p>
                <p>
                    Maior preço negociado:
                </p>
                <p>
                    Maior preço negociado:
                </p>
                <p>
                    Maior preço negociado:
                </p>
                <p>
                    Maior preço negociado:
                </p>

            </div>
        </div>
    );
};


