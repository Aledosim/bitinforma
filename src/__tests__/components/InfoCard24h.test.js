import { enableFetchMocks  } from 'jest-fetch-mock'

enableFetchMocks()
import React from 'react'
import { render, waitFor } from '@testing-library/react'

import { tickerResponse, summaryResponse } from '../fixtures.json'
import CurrencyProvider from '../../contexts/CurrencyContext'

import InfoCard24h from '../../components/InfoCard24h'

describe('<InfoCard24h /> tests', () => {
  beforeEach(() => {
    fetch.resetMocks()

    fetch
      .mockResponse(req => {
        if (/.*\/ticker\/.*/.test(req.url)) {
          return Promise.resolve({ body: JSON.stringify(tickerResponse) })

        } else if (/.*\/day-summary\/.*/.test(req.url)) {
          return Promise.resolve({ body: JSON.stringify(summaryResponse) })

        }
      })
  })

  it('renders without crashing', async () => {

    const tree = render(
      <CurrencyProvider>
        <InfoCard24h />
      </CurrencyProvider>
    )

    // Check if the fetch data is rendered
    await waitFor(() => {
      expect(fetch).toHaveBeenCalledTimes(2)
    })

    expect(tree).toMatchSnapshot()
  })
})
