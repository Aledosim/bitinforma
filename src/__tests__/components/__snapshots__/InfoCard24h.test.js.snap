// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoCard24h /> tests renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="infoCard24hContainer"
        data-cy="infocard24h"
      >
        <header
          class="header"
        >
          <div>
            <div
              class="logoContainer"
              data-cy="coinlogo"
            >
              <span>
                <img
                  alt="coin logo"
                  src="btc.svg"
                />
              </span>
              <span>
                Bitcoin
              </span>
            </div>
          </div>
          <div>
            Nas ultimas 24h
          </div>
        </header>
        <div
          class="content"
        >
          <p>
            <span>
              <img
                alt="seta para cima"
                src="up_arrow.svg"
              />
            </span>
            <span>
              Maior preço negociado:
            </span>
            <span>
              380000.00000000
            </span>
          </p>
          <p>
            <span>
              <img
                alt="seta para baixo"
                src="down_arrow.svg"
              />
            </span>
            <span>
              Menor preço negociado:
            </span>
            <span>
              353684.14000000
            </span>
          </p>
          <p>
            <span>
              <img
                alt="etiqueta de preço"
                src="price_tag.svg"
              />
            </span>
            <span>
              Preço da última negociação:
            </span>
            <span>
              359999.99006000
            </span>
          </p>
          <p>
            <span>
              <img
                alt="preço subindo"
                src="price_up.svg"
              />
            </span>
            <span>
              Menor preço de oferta de venda:
            </span>
            <span>
              360000.00000000
            </span>
          </p>
          <p>
            <span>
              <img
                alt="preço descendo"
                src="price_down.svg"
              />
            </span>
            <span>
              Maior preço de oferta de compra:
            </span>
            <span>
              359999.99006000
            </span>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="infoCard24hContainer"
      data-cy="infocard24h"
    >
      <header
        class="header"
      >
        <div>
          <div
            class="logoContainer"
            data-cy="coinlogo"
          >
            <span>
              <img
                alt="coin logo"
                src="btc.svg"
              />
            </span>
            <span>
              Bitcoin
            </span>
          </div>
        </div>
        <div>
          Nas ultimas 24h
        </div>
      </header>
      <div
        class="content"
      >
        <p>
          <span>
            <img
              alt="seta para cima"
              src="up_arrow.svg"
            />
          </span>
          <span>
            Maior preço negociado:
          </span>
          <span>
            380000.00000000
          </span>
        </p>
        <p>
          <span>
            <img
              alt="seta para baixo"
              src="down_arrow.svg"
            />
          </span>
          <span>
            Menor preço negociado:
          </span>
          <span>
            353684.14000000
          </span>
        </p>
        <p>
          <span>
            <img
              alt="etiqueta de preço"
              src="price_tag.svg"
            />
          </span>
          <span>
            Preço da última negociação:
          </span>
          <span>
            359999.99006000
          </span>
        </p>
        <p>
          <span>
            <img
              alt="preço subindo"
              src="price_up.svg"
            />
          </span>
          <span>
            Menor preço de oferta de venda:
          </span>
          <span>
            360000.00000000
          </span>
        </p>
        <p>
          <span>
            <img
              alt="preço descendo"
              src="price_down.svg"
            />
          </span>
          <span>
            Maior preço de oferta de compra:
          </span>
          <span>
            359999.99006000
          </span>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
